src/tequila/circuit/visualize.py

```diff
@@ -0,0 +1,156 @@
+import tequila
+from typing import Optional, List
+import matplotlib.pyplot as plt
```

Comment:
test fails if matplotlib is not installed (unlikely for most users, buit still). 
would suggest to handle it in the same way as you handle networkx

Reply:
I made the modification so I'll resolve this conversation

Reply:
Hi Rick, the way you handled it in the function itself looks great. There remains the issue that matplotlib and networkx are imported at this point in the file.
This will lead to crashes once we call "import tequila" if those libraries are not implemented.

replace 
```python
import networkx
```
```python
if importlib.util.find_spec("networkx") is not None:
    import networkx
```

and the same for networkx.

The usage of importlib is a good idea, will try to incorporate that in other parts of the code.

Reply:
Ok, I've made changes to visualize.py